study.md

```diff
@@ -21,18 +21,18 @@ There is no requirement to follow any links from these pages.
 After doing all the required readings please tell us anything you were confused about.
 
 ```md
-<!-- your answer here -->
+After doing all the required readings, I am confusing what is different between function declarations and function expressions. And what does hoist meant?
```

Comment:
Here's a good explanation on variable hoisting.  https://en.wikipedia.org/wiki/JavaScript_syntax#Scoping_and_hoisting
Function declarations are hoisted and function expressions have the variable declaration hoisted but not the function definition.  Hopefully this example that I wrote will be helpful.

```javascript
// This will output, "[Function: sayHello]".
console.log(sayHello);

// This will work and output, "Hello".
sayHello();

// This will output, "undefined" since the declaration of "sayHi" has been
// hoisted to the top, but its definition has not been.
console.log(sayHi);

// This will throw an error for the reason above.  Since sayHi is undefined,
// it's trying to execute "undefined" as if it were a function.
sayHi();

// This gets hoisted to the top.
function sayHello() {
  console.log('Hello.');
}

// "var sayHi" gets hoisted to the top, but its assignment to function() { ... }
// does not.
var sayHi = function() {
  console.log('Hi.');
};

// This will output, "[Function: sayHi]", since it's after the definition of the
// function.
console.log(sayHi);

// This will output, "Hi".
sayHi();
```